### PR TITLE
Make Builder more strict

### PR DESCRIPTION
**Pull request by cristaloleg**

### CODE REVIEW

**Review by storozhukBM — APPROVED** [review]
Looks really cool

**Review by junk1tm — APPROVED** [review]
Great, `by default` ftw 😎

File: builq_test.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ func TestBuilder(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			t.Helper()
 			var b Builder
-			b.Addf(format, args...)
+			b.Addf(constString(format), args...)
```
[the remark this answers is not in the document]
Comment (cristaloleg) [reply builq_test.go:33]: Leaving as it's caused diff between 2 options is small.

File: GUIDE.md [hunk 2]
```diff
@@ -8,6 +8,26 @@ Examples in [example_test.go](example_test.go) explicitly show that query argume
...
+// this WILL NOT complile, orClause isn't const
+// var orClause = "OR id = %$"
+// sb.Addf(orClause, 42)
+
+// WILL compile, orClause2 is known at compile-time
+const orClause2 = "OR id = %$"
+sb.Addf(orClause2, 42)
+```
+
+Some projects require constant queries due to security policies (precise definition might be different but you get the idea).
```
[the remark this answers is not in the document]
Comment (cristaloleg) [reply GUIDE.md]: Reworded, thanks.

**Pull request merged**